lib/rucio/web/ui/flask/common/utils.py

```diff
@@ -0,0 +1,626 @@
+    if not token:
+        return render_template("problem.html", msg=msg)
+    return finalize_auth(token, 'x509')
+
+
+def userpass_auth():
+    """
+    Manages login via Rucio USERPASS method.
+    :returns: final page or a page with an error message
+    """
+
+    ui_account = request.args.get('account')
+    ui_vo = request.args.get('vo')
+    username = request.form.get('username')
+    password = request.form.get('password')
+
+    if not username and not password:
+        return render_template('problem.html', msg="No input credentials were provided.")
+
+    if not MULTI_VO:
+        ui_vo = 'def'
+    if ui_account is None and ui_vo is None:
+        ui_account, ui_vo = select_account_name(username, 'userpass', ui_vo)
+    elif ui_account is None:
+        ui_account, _ = select_account_name(username, 'userpass', ui_vo)
+    elif ui_vo is None:
+        _, ui_vo = select_account_name(username, 'userpass', ui_vo)
+
+    if type(ui_vo) is list:
+        if ui_account:
+            valid_vos = []
+            for vo in ui_vo:
+                if account_exists(ui_account, vo):
+                    valid_vos.append(vo)
+
+            if len(valid_vos) == 0:
+                return render_template('problem.html', msg='Cannot find any Rucio account %s associated with identity %s at any VO.' % (html_escape(ui_account), html_escape(username)))
+            elif len(valid_vos) == 1:
+                ui_vo = valid_vos[0]
+            else:
+                vos_with_desc = get_vo_descriptions(valid_vos)
+                return render_template('login.html', account=ui_account, vo=None, possible_vos=vos_with_desc)
+        else:
+            vos_with_desc = get_vo_descriptions(ui_vo)
+            return render_template('login.html', account=None, vo=None, possible_vos=vos_with_desc)
+
+    if not ui_account:
+        if MULTI_VO:
+            return render_template('problem.html', msg='Cannot get find any account associated with %s identity at VO %s.' % (html_escape(username), html_escape(ui_vo)))
+        else:
+            return render_template('problem.html', msg='Cannot get find any account associated with %s identity.' % (html_escape(username)))
+    token = get_token(auth.get_auth_token_user_pass, acc=ui_account, vo=ui_vo, idt=username, pwd=password)
+    if not token:
+        if MULTI_VO:
+            return render_template('problem.html', msg='Cannot get auth token. It is possible that the presented identity %s is not mapped to any Rucio account %s at VO %s.') % (html_escape(username), html_escape(ui_account), html_escape(ui_vo))
+        else:
+            return render_template('problem.html', msg='Cannot get auth token. It is possible that the presented identity %s is not mapped to any Rucio account %s.') % (html_escape(username), html_escape(ui_account))
+
+    return finalize_auth(token, 'userpass')
+
+
+def saml_auth(method, data=None):
+    """
+    # TODO: Validate for Flask
```

Comment:
I guess this should be done before the merge

Reply:
I don't really have a setup to this. It was developed by an external who is not in the project anymore. This is not deployed anywhere, yet.